openstack_dashboard/api/glance.py

```diff
@@ -152,64 +154,42 @@ def __ne__(self, other_image):
-                    return app['id']
-                if(app['chi_tacc_appliance_id'] == self.id):
-                    return app['id']
-                if(app['kvm_tacc_appliance_id'] == self.id):
-                    return app['id']
+            if not Image.PUBLISHED_APPLIANCES.get(self.id):
+                return -1
+
+            return Image.PUBLISHED_APPLIANCES.get(self.id).get('id')
         except:
             LOG.error('Error getting catalog id from appliance catalog api for image id: ' + self.id)
         return -1
 
     def get_is_project_supported(self):
         try:
-            LOG.info('Getting project_supported flag from appliance catalog api for image id: ' + self.id)
-            app_json = json.loads(Image.APP_CATALOG)
-            for app in app_json['result']:
-                if(app['chi_uc_appliance_id'] == self.id):
-                    return app['project_supported']
-                if(app['chi_tacc_appliance_id'] == self.id):
-                    return app['project_supported']
-                if(app['kvm_tacc_appliance_id'] == self.id):
-                    return app['project_supported']
+            if not Image.PUBLISHED_APPLIANCES.get(self.id):
+                return False
+
+            return Image.PUBLISHED_APPLIANCES.get(self.id).get('project_supported')
         except:
             LOG.error('Error getting project_supported flag from appliance catalog api for image id: ' + self.id)
         return False
 
     def get_is_published_in_app_catalog(self):
         try:
-            LOG.info('Checking if image is published to appliance catalog for image id: ' + self.id)
-            app_json = json.loads(Image.APP_CATALOG)
-            for app in app_json['result']:
-                if(app['chi_uc_appliance_id'] == self.id):
-                    return True
-                if(app['chi_tacc_appliance_id'] == self.id):
-                    return True
-                if(app['kvm_tacc_appliance_id'] == self.id):
-                    return True
+            return Image.PUBLISHED_APPLIANCES.get(self.id) is not None
         except:
-            LOG.error('Error checking if image is published to appliance catalog for image id: ' + self.id)
+             LOG.error('Error checking if image is published to appliance catalog for image id: ' + self.id)
         return False
 
-@memoized
-def fetch_supported_appliances(request):
-    try:
-        LOG.info('Fetching and caching Appliance JSON from ' + settings.CHAMELEON_PORTAL_API_BASE_URL + settings.APPLIANCE_CATALOG_API_PATH)
-        http = urllib3.PoolManager()
-        r = http.request('GET', settings.CHAMELEON_PORTAL_API_BASE_URL + settings.APPLIANCE_CATALOG_API_PATH)
-        LOG.debug('fetched appliance catalog data: ' + r.data)
-        return r.data
-    except Exception as e:
-        LOG.error(e)
-    return []
+def fetch_published_appliances(request):
+    if not cache.get('app_catalog_updated'):
```

Comment:
I think this is probably where the biggest gain came from! The `@memoized` function cached the return value of this function based on the input args. Normally that works fine, but in this case the `request` is always going to be a bit different (even b/c of something as simple as a timestamp on the request), so in practice I don't think it was caching anything even if you reloaded the page quickly.

Reply:
`@memoized` definitely wasn't much help there